server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -2141,6 +2141,11 @@ IndexMetadata build(boolean repair) {
                 // to be able to build a temporary instance
                 tierPreference = null;
             }
+            if (SearchableSnapshotsSettings.isPartialSearchableSnapshotIndex(settings)
+                && (tierPreference == null || tierPreference.size() != 1 || tierPreference.get(0).equals(DataTier.DATA_FROZEN) == false)) {
+                settings = Settings.builder().put(settings).put(DataTier.TIER_PREFERENCE, DataTier.DATA_FROZEN).build();
```

Comment:
Does this need to increment the `settingsVersion`, since a change is being made?

Reply:
That's a great call. I _think_ so. I'm wondering if it'd matter if a new index starts with version 2 though? This  code path is shared for new and restored indices.

I don't think it does (or it shouldn't).